src/femr/labelers/omop_inpatient_admissions.py

```diff
@@ -109,17 +109,14 @@ def get_outcome_times(self, patient: Patient) -> List[datetime.datetime]:
     def get_prediction_times(self, patient: Patient) -> List[datetime.datetime]:
         """Return end of admission as prediction timm."""
         times: List[datetime.datetime] = []
-        prev_discharge_date: datetime.date = datetime.date(1900, 1, 1)
-
+        admission_times = set()
         for admission_time, discharge_time in get_inpatient_admission_discharge_times(patient, self.ontology):
             prediction_time: datetime.datetime = self.prediction_time_adjustment_func(discharge_time)
-
             # Ignore patients who are readmitted the same day they were discharged b/c of data leakage
-            if admission_time.date() <= prev_discharge_date:
+            if prediction_time.replace(hour=0, minute=0, second=0, microsecond=0) in admission_times:
```

Comment:
Why was this change made?

Reply:
I think both implementations are slightly off actually on second look -- have re-implemented what I think is correct